l10n_it_fatturapa_in/wizard/wizard_import_fatturapa.py

```diff
@@ -1888,8 +1881,52 @@ def importFatturaPA(self):
                     )
                     invoice.write({"tax_representative_id": tax_partner_id})
                 if Intermediary:
-                    Intermediary_id = self.getPartnerBase(Intermediary.DatiAnagrafici)
-                    invoice.write({"intermediary": Intermediary_id})
+                    vat = False
+                    nome = Intermediary.DatiAnagrafici.Anagrafica.Nome or ""
+                    cognome = Intermediary.DatiAnagrafici.Anagrafica.Cognome or ""
+                    denominazione = (
+                        Intermediary.DatiAnagrafici.Anagrafica.Denominazione or ""
+                    )
+
+                    if Intermediary.DatiAnagrafici.IdFiscaleIVA:
+                        partner = self.getPartnerBase(
+                            Intermediary.DatiAnagrafici, partnerCreate=False
+                        )
+
+                        if isinstance(partner, int):
+                            partner_id = self.env["res.partner"].browse(partner)
+                            invoice.write({"intermediary": partner_id})
+                        elif isinstance(partner, dict):
+                            partner_vals = partner
+                            new_intermediary_data.append(
+                                {
+                                    "name": partner_vals.get("name"),
+                                    "firstname": partner_vals.get("firstname"),
+                                    "lastname": partner_vals.get("lastname"),
+                                    "vat": partner_vals["vat"],
+                                    "invoice_id": invoice.id,
+                                    "intermediary_data_json": json.dumps(partner_vals),
+                                }
+                            )
+                        elif isinstance(partner, bool):
+                            invoice.write({"intermediary": False})
+                    else:
+                        new_intermediary_data.append(
+                            {
+                                "name": denominazione,
+                                "firstname": nome,
+                                "lastname": cognome,
+                                "vat": vat,
+                                "invoice_id": invoice.id,
+                                "intermediary_data_json": {
+                                    "name": denominazione,
+                                    "firstname": nome,
+                                    "lastname": cognome,
+                                    "vat": vat,
+                                },
```

Comment:
```suggestion
                                "invoice_id": invoice.id,
                                "intermediary_data_json": {
                                    "name": denominazione,
                                    "firstname": nome,
                                    "lastname": cognome,
                                    "vat": vat,
                                },
```
I valori dei primi campi potrebbero essere calcolati in base a quello che c'è in `intermediary_data_json` invece di ripeterli.

Reply:
È necessario allora implementare una funzione in `WizardCheckIntermediaryVatLine` che estragga i dati da `intermediary_data_json`